src/definitions.jl

```diff
@@ -680,15 +680,31 @@ adjoint_mul(p::Plan, x::AbstractArray) = adjoint_mul(p, x, AdjointStyle(p))
 function adjoint_mul(p::Plan{T}, x::AbstractArray, ::FFTAdjointStyle) where {T}
     dims = fftdims(p)
     N = normalization(T, size(p), dims)
-    return (p \ x) / N
+    pinv = inv(p)
+    # Optimization: when pinv is a ScaledPlan, check if we can avoid a loop over x.  
+    # Even if not, ensure that we do only one pass by combining the normalization with the plan.
+    if pinv isa ScaledPlan && isapprox(pinv.scale, N)
```

Comment:
```suggestion
    if pinv isa ScaledPlan && pinv.scale == N
```

`isapprox` only compares half of the significant digits, which seems unsafe.

Why not just use `==` here?  The `inv(p)` scaling is ordinarily computed by calling `normalization` (and even if not, as long as it is computed by `1/N` it should give the same result).  Moreover, even if `==` returns `false`, the worse that happens is that we do an unnecessary scaling by something that is `≈ 1`.